main.go

```diff
@@ -48,14 +52,33 @@ func main() {
 		locationBaseURL: *locationBaseURL0,
 		filesDir:        *dataDir0 + "/files/",
 		metaDir:         *dataDir0 + "/meta/",
+		port:            *port0,
+	}
+
+	if opt.locationBaseURL == "" {
+		logrus.Error("'--location-base-url' is required")
+		os.Exit(1)
 	}
 
 	if !strings.HasPrefix(opt.locationBaseURL, "http://") && !strings.HasPrefix(opt.locationBaseURL, "https://") {
-		logrus.Errorf("'--location-base-url' is required and must be in format 'http://[host]:[port] or https://[host]:[port]'")
+		logrus.Error("'--location-base-url' must be in format 'http://[host] or https://[host]'")
+		os.Exit(1)
+	}
+
+	u, err := url.Parse(opt.locationBaseURL)
+	if err != nil {
+		logrus.Errorf("Invalid '--location-base-url': %v", err)
 		os.Exit(1)
 	}
 
-	opt.locationBaseURL = strings.Trim(opt.locationBaseURL, "/")
+	// Add port to the base URL if not already present
+	if u.Port() == "" {
```

Comment:
The port could be gotten from the url as the url spec can handle port number definitions already